src/transformers/models/mamba2/modeling_mamba2.py

```diff
@@ -964,8 +964,8 @@ def prepare_inputs_for_generation(
         attention_mask: Optional[torch.Tensor] = None,
         **kwargs,
     ):
-        if input_ids.shape[1] == 0:
-            past_len = inputs_embeds.shape[1]
+        if inputs_embeds is not None:
+            past_len = inputs_embeds.shape[1] + input_ids.shape[1]
```

Comment:
this is no longer the past length, but the current length 😉 Otherwise as this is used to create the cache positions, this is valid 😉

Reply:
Fair enough :p